src/main/kotlin/no/nav/sbl/sosialhjelp_mock_alt/otherEndpoints/frontend/model/FrontendPersonalia.kt

```diff
@@ -259,36 +247,21 @@ data class FrontendUtbetalingFraNav(
     }
 
     companion object {
-        fun mapUtbetalingDtoToFrontendUtbetalingFraNav(utbetaling: UtbetalingDto): FrontendUtbetalingFraNav {
-            return FrontendUtbetalingFraNav(
-                utbetaling.netto,
-                utbetaling.utbetalingsdato ?: LocalDate.now(),
-                utbetaling.tittel,
-                "", // melding?
-                utbetaling.skattetrekk,
-                utbetaling.komponenter.first().type ?: ""
-            )
-        }
+        fun mapUtbetalingDtoListeTilFrontendUtbetalingerFraNavListe(utbetalingerFraNav: List<UtbetalDataDto>): List<FrontendUtbetalingFraNav> {
 
-        // TODO: bytt ut metode over med denne når vi har skrudd over til utbetalData. Pt ubrukt.
-        fun mapUtbetalDataDtoToFrontendUtbelingFraNav(utbetalData: UtbetalDataDto): FrontendUtbetalingFraNav {
-
-            // TODO  en del verdier her som er én verdi kommer fra lister. Velger nå første i liste, men hvordan skal logikken her henge sammen?
-
-            var skattsum = 0.00
-
-            for (ytelse in utbetalData.ytelseListe!!) {
-                skattsum += ytelse.skattsum?.toDouble() ?: 0.0
+            return utbetalingerFraNav.flatMap { utbetalingFraNav ->
+                utbetalingFraNav.ytelseListe
+                    ?.map {
+                        FrontendUtbetalingFraNav(
+                            belop = it.ytelseNettobeloep?.toDouble() ?: 0.00,
+                            dato = utbetalingFraNav.utbetalingsdato ?: LocalDate.now(),
+                            ytelsestype = it.ytelsestype ?: "",
+                            melding = utbetalingFraNav.utbetalingsmelding ?: "",
+                            skattebelop = it.skattsum?.toDouble() ?: 0.00,
+                            ytelseskomponenttype = it.ytelseskomponentListe?.first()?.ytelseskomponenttype ?: ""
```

Comment:
ytelseskomponenttype benyttes ikke i søknad. Litt usikker på hvorfor den er med i mock. Men siden den ikke benyttes i søknad, tok jeg sjansen på å kun hente det fra evt første element i komponentlisten. Men vi kan kanskje vurdere å ta det helt bort? Da må isåfall frontend også oppdateres.

Reply:
Godt spørsmål. Hvis det ikke benyttes i det hele tatt i søknaden, ser jeg ikke noen grunn til at det burde være med her i mock.

Reply:
Lager en egen sak på det og merger denne!